tools/devtoolsNG/home.html

```diff
@@ -0,0 +1,41 @@
+<body>
+
+  <div id="log"></div> 
+ <input type="button" onclick="myFunction()" value="Clear">
+
+ <script>
+
+   const websocket = new WebSocket('ws://localhost:12345');
+
+   websocket.onopen = e => {
+
+    websocket.send('Hello dear Server!');
+
+     websocket.onmessage = msg => {
+
+      document.getElementById("log").innerHTML = document.getElementById("log").innerHTML + "<div>" + msg.data + "</div>"

```

Comment:
Just FYI: this is kind of bad, because you force the browser to re-render the entire page on every element.

The proper-er way to do this would be something like:
```
const newEntry = document.createElement('div');
newEntry.innerText = msg.data
document.getElementById("log").append(newEntry);
```

It doesn't really matter, but FYI;